python/pyspark/rdd.py

```diff
@@ -869,7 +871,10 @@ def collect(self):
             to be small, as all the data is loaded into the driver's memory.
         """
         with SCCallSiteSync(self.context) as css:
-            sock_info = self.ctx._jvm.PythonRDD.collectAndServe(self._jrdd.rdd())
+            java_map = MapConverter().convert(self.context.getLocalProperties(),
+                                              self.context._gateway._gateway_client)
+            sock_info = self.ctx._jvm.PythonRDD.collectAndServe(
+                self._jrdd.rdd() ,java_map)
```

Comment:
There are multiple actions like `toPandas`. If the changes are needed to all of them, it sounds like we're re-implementing the local property logic within Python side.